Source/OnboardingContentViewController.m

```diff
@@ -29,10 +30,17 @@
 
 @interface OnboardingContentViewController ()
 
+@property (nonatomic, strong) MPMoviePlayerController *moviePlayerController;
+@property (nonatomic, strong) NSURL *videoURL;
+
 @end
 
 @implementation OnboardingContentViewController
 
+- (void)dealloc {
+    [self removeObserver:self forKeyPath:UIApplicationWillEnterForegroundNotification];
```

Comment:
I've never actually used this version of `removeObserver:`, meaning the one with `keyPath:`. I've always used `[[NSNotificationCenter defaultCenter] removeObserver:self];`. Does this work the same way?